components/bme280/src/bme280.c

```diff
@@ -0,0 +1,309 @@
+void i2c_master_init(void)
+{
+    // Configuration structure for I2C
+    i2c_config_t i2c_config = {
+        .mode = I2C_MODE_MASTER,                // I2C mode: Master
+        .sda_io_num = CONFIG_I2C_MASTER_SDA_IO, // GPIO pin for SDA (data line)
+        .scl_io_num = CONFIG_I2C_MASTER_SCL_IO, // GPIO pin for SCL (clock line)
+        .sda_pullup_en = GPIO_PULLUP_ENABLE,    // Enable SDA pull-up resistor
+        .scl_pullup_en = GPIO_PULLUP_ENABLE,    // Enable SCL pull-up resistor
+        .master.clk_speed = 1000000             // Clock speed (1 MHz)
+    };
+
+    // Configure I2C parameters
+    i2c_param_config(I2C_NUM_0, &i2c_config);
+
+    // Install the I2C driver
+    i2c_driver_install(I2C_NUM_0, I2C_MODE_MASTER, 0, 0, 0);
+}
+
+int8_t i2c_write(uint8_t dev_addr, uint8_t reg_addr, uint8_t *reg_data, uint8_t cnt)
+{
+    int8_t iError = SUCCESS;
+    esp_err_t espRc;
+
+    // Create I2C command link
+    i2c_cmd_handle_t cmd = i2c_cmd_link_create();
+
+    // Write device address and register address for writing
+    i2c_master_start(cmd);
+    i2c_master_write_byte(cmd, (dev_addr << 1) | I2C_MASTER_WRITE, true);
+    i2c_master_write_byte(cmd, reg_addr, true);
+
+    // Write data to the device
+    i2c_master_write(cmd, reg_data, cnt, true);
+
+    // Stop the I2C communication
+    i2c_master_stop(cmd);
+
+    // Execute the I2C command sequence
+    espRc = i2c_master_cmd_begin(I2C_NUM_0, cmd, 10 / portTICK_PERIOD_MS);
+    if (espRc == ESP_OK)
+    {
+        iError = 0; // Success
+    }
+    else
+    {
+        iError = -1; // Error
+    }
+
+    // Delete the I2C command link to free up resources
+    i2c_cmd_link_delete(cmd);
+
+    return iError; // Return the result
+}
+
+int8_t i2c_read(uint8_t dev_addr, uint8_t reg_addr, uint8_t *reg_data, uint8_t cnt)
+{
+    int8_t iError = SUCCESS;
+    esp_err_t espRc;
+
+    // Create I2C command link
+    i2c_cmd_handle_t cmd = i2c_cmd_link_create();
+
+    // Write device address and register address for reading
+    i2c_master_start(cmd);
+    i2c_master_write_byte(cmd, (dev_addr << 1) | I2C_MASTER_WRITE, true);
+    i2c_master_write_byte(cmd, reg_addr, true);
+
+    // Start a new I2C communication for reading
+    i2c_master_start(cmd);
+    i2c_master_write_byte(cmd, (dev_addr << 1) | I2C_MASTER_READ, true);
+
+    // Read data from the device
+    if (cnt > 1)
+    {
+        i2c_master_read(cmd, reg_data, cnt - 1, I2C_MASTER_ACK);
+    }
+    i2c_master_read_byte(cmd, reg_data + cnt - 1, I2C_MASTER_NACK);
+
+    // Stop the I2C communication
+    i2c_master_stop(cmd);
+
+    // Execute the I2C command sequence
+    espRc = i2c_master_cmd_begin(I2C_NUM_0, cmd, 10 / portTICK_PERIOD_MS);
+    if (espRc == ESP_OK)
+    {
+        iError = 0; // Success
+    }
+    else
+    {
+        iError = -1; // Error
+    }
+
+    // Delete the I2C command link to free up resources
+    i2c_cmd_link_delete(cmd);
+
+    return iError; // Return the result
```

Comment:
Some inconsistency in variables naming style, here `iError` and ex in bme280_init_driver `com_rslt`. Rather do not mix camel case with snake case. Please check other places in code as well.